Formula/libsigrok.rb

```diff
@@ -78,15 +74,25 @@ class Libsigrok < Formula
 
   def install
     resource("fw-fx2lafw").stage do
-      system "./autogen.sh"
+      if File.exist?("autogen.sh")
+        system "./autogen.sh"
+      else
+        system "autoreconf", "-fiv"
+      end
```

Comment:
It's never there in the tarbal so no need to do an if here.

Reply:
`autogen.sh` is only present in the Git repositories (not the tarballs) but the conditionals are necessary if we want `brew install --HEAD` to continue to work like the existing formula setup. Granted, [`autogen.sh` basically just does `autoreconf -fiv`](https://sigrok.org/gitweb/?p=libsigrok.git;a=blob;f=autogen.sh;h=09de1ba5f467225dc5e78423f0a6039a0f433ad3;hb=HEAD) at the moment but I figured it may be better to err on the side of caution (e.g., in case upstream changes this setup in the future). This would also automatically use an `autogen.sh` file if they appear in tarballs in the future.

If you don't think it's worth the additional verbosity, I'll remove it (as `system "autoreconf", "-fiv"` works for both `stable` and `head` builds).

Reply:
`if build.head?` is typically used to handle this situation but it's less flexible than the `if File.exist?("autogen.sh")` approach here. If the Git repository removes `autogen.sh` in the future, head builds would break and the formula would have to be modified. Similarly, if the tarballs start including `autogen.sh` in the future, stable builds wouldn't use it unless someone notices and the formula is modified.

If we need to use if/else here either way (i.e., same number of lines), wouldn't it be better to approach this in a way that would appropriately handle this situation regardless of whether it's a stable/head build? To put it another way, shouldn't the condition directly reflect what we're concerned with, the existence of `autogen.sh`, rather than the context where the file is assumed to be present? Some assumptions are unavoidable but this one doesn't seem necessary.

I can understand if the answer is simply, "These potential scenarios aren't compelling, we can cross those bridges if/when we come to them, and this should fall in line with other formulae (i.e., using `if build.head?`)" but I wanted to highlight the potential shortcomings in case we're interested in reevaluating this approach for the ~150 similar formulae.

Reply:
We're not trying to define an all encompassing build script for libsigrok. We just want to make it build in homebrew at this moment.

I'd even say that if the tarbal contents change we want it to break since otherwise we would lack a clear indication of the change.

Reply:
> I'd even say that if the tarbal contents change we want it to break since otherwise we would lack a clear indication of the change.

The thought crossed my mind when I was considering potential motivations behind using a less flexible approach, so I can understand this stance.

Thanks for discussing. I'll update this to use `if build.head?` in a moment.